packages/main/src/FileUploader.ts

```diff
@@ -317,6 +357,28 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Returns true if all files are below the maxFileSize, otherwise throws a "fileSizeExceeded" event if one file exceeds it.
+	 * @private
+	 */
+	_checkFileSize(files: FileList | null): boolean {
+		if (!(this.maxFileSize && files)) {
+			return true;
+		}
+
+		for (let i = 0; i < files.length; i++) {
+			const fileSize = (files[i].size / 1024) / 1024;
+			if (fileSize > this.maxFileSize) {
+				this.fireEvent<FileUploaderFileSizeExceededEventDetail>("fileSizeExceeded", {
```

Comment:
I would also fire the event outside this method, cause now the method is doing 2 things, checking as the name says and also fires an event. As a boolean check function I would only expect a result, not any actions. Return the event based on the result of the function.

Reply:
As seperately discussed, this isn't possible because the event needs the file name and calculated size. Instead, as we agreed in private, I changed the function name to not be so specific.